ee/session_recordings/playlist_counters/recordings_that_match_playlist_filters.py

```diff
@@ -458,14 +490,21 @@ def count_recordings_that_match_playlist_filters(playlist_id: int) -> None:
 
 
 def enqueue_recordings_that_match_playlist_filters() -> None:
-    all_playlists = (
-        SessionRecordingPlaylist.objects.filter(
-            deleted=False,
-            filters__isnull=False,
-        )
-        .filter(Q(last_counted_at__isnull=True) | Q(last_counted_at__lt=timezone.now() - timedelta(hours=2)))
-        .order_by(F("last_counted_at").asc(nulls_first=True))[: settings.PLAYLIST_COUNTER_PROCESSING_PLAYLISTS_LIMIT]
-    )
+    base_query = SessionRecordingPlaylist.objects.filter(
+        deleted=False,
+        filters__isnull=False,
+    ).filter(Q(last_counted_at__isnull=True) | Q(last_counted_at__lt=timezone.now() - timedelta(hours=2)))
+
+    total_playlists_count = base_query.count()
```

Comment:
django orm's `count` only loads the count, so it doesn't stand the overhead of creating the models

Reply:
Why `count` before though? Coluld be missing the context, but if you still get `all_playlists`, why not count them and avoid the first count call?

Reply:
we don't get all we call (effectively) `base_query[:limit]` which means we only load up to the limit so `len(results)` would never be greater than limit

so we have to count to get the total

(or load them all into memory and _then_ slice them but since we don't know how many there will be over time it feels safer to count an slice separately)